src/java/org/apache/cassandra/cql3/terms/Constants.java

```diff
@@ -499,7 +509,9 @@ else if (column.type instanceof NumberType<?>)
             else if (column.type instanceof StringType)
             {
                 ByteBuffer append = t.bindAndGet(params.options);
-                ByteBuffer current = getCurrentCellBuffer(partitionKey, params);
+                if (append == null)
```

Comment:
would NPE with `+= null`

Reply:
Is `+=` null an error or should it silently succeed and do nothing? How do we get to appending `null`? I am not clear on what happens here.